Formula/f3d.rb

```diff
@@ -0,0 +1,39 @@
+class F3d < Formula
+  desc "Fast and minimalist 3D viewer"
+  homepage "https://kitware.github.io/F3D/"
+  url "https://gitlab.kitware.com/f3d/f3d/-/archive/v1.0.1/f3d-v1.0.1.tar.gz"
+  sha256 "fb362dba3ccf49db9e8841d8a5310f37399bfea8866b6e040ce85670d54b97f7"
+  license "BSD-3-Clause"
+
+  depends_on "cmake" => :build
+  depends_on "vtk"
+
+  def install
+    # Is fixed upstream, remove this line for v1.1
+    inreplace "src/F3DOptions.cxx", "/etc/f3d/", "/usr/local/etc/f3d/"
+
+    args = std_cmake_args + %W[
+      -DCMAKE_BUILD_TYPE:STRING=Release
+      -DMACOSX_BUILD_BUNDLE:BOOL=OFF
+      -DBUILD_SHARED_LIBS:BOOL=ON
+      -DBUILD_TESTING:BOOL=OFF
+      -DF3D_INSTALL_DEFAULT_CONFIGURATION_FILE:BOOL=ON
+      -DCMAKE_INSTALL_NAME_DIR:STRING=#{lib}
+      -DCMAKE_INSTALL_RPATH:STRING=#{lib}
+    ]
+
+    mkdir "build" do
+      system "cmake", "..", *args
+      system "make"
+      system "make", "install"
+    end
+
+    mkdir_p prefix/"f3d"
+    mv prefix/"config.json", prefix/"f3d/config.json"
+    etc.install_symlink prefix/"f3d"
+  end
+
+  test do
+    system "#{bin}/f3d", "--version"
```

Comment:
Is there any better than just test the version?

Reply:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
yes I will improve that. There is currently no 3D file format in the test fixtures. I can manually create one in the formula but maybe it makes sense to add one for other software? (in OBJ format for example which is quite standard)

Reply:
I have created a very basic OBJ file in the test.